src/ReportsSync.php

```diff
@@ -208,669 +51,116 @@ public static function get_import_totals( $days, $skip_existing ) {
-				self::QUEUE_BATCH_ACTION,
-				self::QUEUE_DEPENDENT_ACTION,
-				self::CUSTOMERS_IMPORT_BATCH_ACTION,
-				self::CUSTOMERS_DELETE_BATCH_INIT,
-				self::CUSTOMERS_DELETE_BATCH_ACTION,
-				self::ORDERS_IMPORT_BATCH_ACTION,
-				self::ORDERS_IMPORT_BATCH_INIT,
-				self::ORDERS_DELETE_BATCH_INIT,
-				self::ORDERS_DELETE_BATCH_ACTION,
-				self::SINGLE_CUSTOMER_IMPORT_ACTION,
-				self::SINGLE_ORDER_IMPORT_ACTION,
-			);
-			$store->clear_pending_wcadmin_actions( $action_types );
-		} elseif ( version_compare( \ActionScheduler_Versions::instance()->latest_version(), '3.0', '>=' ) ) {
-			$store->cancel_actions_by_group( self::QUEUE_GROUP );
-		} else {
-			self::queue()->cancel_all( null, array(), self::QUEUE_GROUP );
-		}
-	}
-
-	/**
-	 * Delete all data for reports.
-	 *
-	 * @return string
-	 */
-	public static function delete_report_data() {
-		// Cancel all pending import jobs.
-		self::clear_queued_actions();
-
-		// Delete orders in batches.
-		self::queue()->schedule_single( time() + 5, self::ORDERS_DELETE_BATCH_INIT, array(), self::QUEUE_GROUP );
-
-		// Delete customers after order data is deleted.
-		self::queue_dependent_action( self::CUSTOMERS_DELETE_BATCH_INIT, array(), self::ORDERS_DELETE_BATCH_INIT );
-
-		// Delete import options.
-		delete_option( 'wc_admin_import_customers_count' );
-		delete_option( 'wc_admin_import_orders_count' );
-		delete_option( 'wc_admin_import_customers_total' );
-		delete_option( 'wc_admin_import_orders_total' );
-		delete_option( 'wc_admin_imported_from_date' );
-
-		return __( 'Report table data is being deleted.', 'woocommerce-admin' );
-	}
-
-	/**
-	 * Schedule an action to import a single Order.
-	 *
-	 * @param int $order_id Order ID.
-	 * @return void
-	 */
-	public static function schedule_single_order_import( $order_id ) {
-		if ( 'shop_order' !== get_post_type( $order_id ) && 'woocommerce_refund_created' !== current_filter() ) {
-			return;
-		}
-
-		if ( apply_filters( 'woocommerce_analytics_disable_order_scheduling', false ) ) {
-			self::orders_lookup_import_order( $order_id );
-			return;
-		}
-
-		// This can get called multiple times for a single order, so we look
-		// for existing pending jobs for the same order to avoid duplicating efforts.
-		$existing_jobs = self::queue()->search(
-			array(
-				'status'   => 'pending',
-				'per_page' => 1,
-				'claimed'  => false,
-				'search'   => "[{$order_id}]",
-				'group'    => self::QUEUE_GROUP,
-			)
-		);
-
-		if ( $existing_jobs ) {
-			$existing_job = current( $existing_jobs );
-
-			// Bail out if there's a pending single order action, or a pending dependent action.
-			if (
-				( self::SINGLE_ORDER_IMPORT_ACTION === $existing_job->get_hook() ) ||
-				(
-					self::QUEUE_DEPENDENT_ACTION === $existing_job->get_hook() &&
-					in_array( self::SINGLE_ORDER_IMPORT_ACTION, $existing_job->get_args(), true )
-				)
-			) {
-				return;
+		foreach ( self::get_schedulers() as $scheduler ) {
+			if ( $scheduler::is_importing() ) {
```

Comment:
Same question here about checking if `is_importing` is callable (re: filter). Optionally checking to see if the trait is being used if you think that's more robust (and possible).

Reply:
Resolved via https://github.com/woocommerce/woocommerce-admin/pull/3285/commits/e7fd4e813f7c3618fd07d33243930e801c008a9e